reckoner/tests/test_cli.py

```diff
@@ -0,0 +1,59 @@
+'''Testing of CLI commands in click, contract tests'''
+
+import unittest
+import mock
+from click.testing import CliRunner
+from reckoner import cli
+
+
+class TestCli(unittest.TestCase):
+    '''
+    Test all the contracts of the command line tool.
+    If something changes in here than you should consider that in your version
+    bumping scheme.
+    '''
+
+    def test_version(self):
+        '''Assure version subcommand exists'''
+        runner = CliRunner()
+        result = runner.invoke(cli.version)
+
+        self.assertEqual(0, result.exit_code)
+        self.assertNotEqual('', result.output)
+
+        result = runner.invoke(cli.cli, args=['--version'])
+
+        self.assertEqual(0, result.exit_code)
+        self.assertNotEqual('', result.output)
+
+    def test_supports_loglevel(self):
+        '''assure we have a global command for --log-level'''
+        runner = CliRunner()
+        result = runner.invoke(
+            cli.cli, args=['--log-level', 'DEBUG', 'version'])
```

Comment:
This test kinda sucks, there's probably a better way to aggregate available parameters.